From 5867adc76a33e93b8709b11eca48fc2c902d5c9f Mon Sep 17 00:00:00 2001
From: Mark Adler <madler@alumni.caltech.edu>
Date: Sun, 11 Dec 2016 21:53:09 -0800
Subject: [PATCH] Add -M, --time option to store or restore the modification
 time.

This replaces the previous -M for --maxsplits with -J. (Hopefully
no one was using -M. Probably not.) This fills a hole left by the
change of -n, --no-name to also not store/restore the modification
time. Now -nM will store/restore the modification time, but not
the name. The full set of combinations is -n for neither, -N for
both, -nM for just the time, and -Nm for just the name. Note that
the order is important. -n or -N will overwrite the effect of a
preceding -m or -M. The default for compression is -N, and the
default for decompression is -n. So -N or -n can be left off for
those cases, respectively.
---
 pigz.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/pigz.c b/pigz.c
index 5b48f5c..6ae2c96 100644
--- a/pigz.c
+++ b/pigz.c
@@ -3795,12 +3795,13 @@ local char *helptext[] = {
 "  -h, --help           Display a help screen and quit",
 "  -i, --independent    Compress blocks independently for damage recovery",
 "  -I, --iterations n   Number of iterations for -11 optimization",
+"  -J, --maxsplits n    Maximum number of split blocks for -11",
 "  -k, --keep           Do not delete original file after processing",
 "  -K, --zip            Compress to PKWare zip (.zip) single entry format",
 "  -l, --list           List the contents of the compressed input",
 "  -L, --license        Display the pigz license and quit",
 "  -m, --no-time        Do not store or restore mod time",
-"  -M, --maxsplits n    Maximum number of split blocks for -11",
+"  -M, --time           Store or restore mod time",
 "  -n, --no-name        Do not store or restore file name or mod time",
 "  -N, --name           Store or restore file name and mod time",
 "  -O  --oneblock       Do not split into smaller blocks for -11",
@@ -3899,11 +3900,11 @@ local char *longopts[][2] = {
     {"LZW", "Z"}, {"ascii", "a"}, {"best", "9"}, {"bits", "Z"},
     {"blocksize", "b"}, {"decompress", "d"}, {"fast", "1"}, {"first", "F"},
     {"force", "f"}, {"help", "h"}, {"independent", "i"}, {"iterations", "I"},
-    {"keep", "k"}, {"license", "L"}, {"list", "l"}, {"maxsplits", "M"},
+    {"keep", "k"}, {"license", "L"}, {"list", "l"}, {"maxsplits", "J"},
     {"name", "N"}, {"no-name", "n"}, {"no-time", "m"}, {"oneblock", "O"},
     {"processes", "p"}, {"quiet", "q"}, {"recursive", "r"}, {"rsyncable", "R"},
     {"silent", "q"}, {"stdout", "c"}, {"suffix", "S"}, {"test", "t"},
-    {"to-stdout", "c"}, {"uncompress", "d"}, {"verbose", "v"},
+    {"time", "M"}, {"to-stdout", "c"}, {"uncompress", "d"}, {"verbose", "v"},
     {"version", "V"}, {"zip", "K"}, {"zlib", "z"}};
 #define NLOPTS (sizeof(longopts) / (sizeof(char *) << 1))
 
@@ -3997,6 +3998,7 @@ local int option(char *arg)
                 break;
             case 'F':  g.zopts.blocksplittinglast = 1;  break;
             case 'I':  get = 4;  break;
+            case 'J':  get = 5;  break;
             case 'K':  g.form = 2;  g.sufx = ".zip";  break;
             case 'L':
                 fputs(VERSION, stderr);
@@ -4005,7 +4007,7 @@ local int option(char *arg)
                       stderr);
                 fputs("No warranty is provided or implied.\n", stderr);
                 exit(0);
-            case 'M':  get = 5;  break;
+            case 'M':  g.headis |= 0xa;  break;
             case 'N':  g.headis = 0xf;  break;
             case 'O':  g.zopts.blocksplitting = 0;  break;
             case 'R':  g.rsync = 1;  break;
@@ -4047,7 +4049,7 @@ local int option(char *arg)
             return 0;
     }
 
-    /* process option parameter for -b, -p, -S, -I, or -M */
+    /* process option parameter for -b, -p, -S, -I, or -J */
     if (get) {
         size_t n;
 
-- 
2.11.0

